packages/inspection-capture-web/src/PhotoCapture/hooks/usePhotoCaptureTutorial.ts

```diff
@@ -0,0 +1,86 @@
+import { useEffect, useState } from 'react';
+import { CaptureAppConfig, PhotoCaptureTutorialOption } from '@monkvision/types';
+import { useObjectMemo } from '@monkvision/common';
+
+export const STORAGE_KEY_PHOTO_CAPTURE_TUTORIAL = '@monk_photoCaptureTutorial';
+
+/**
+ * Enum of the different steps of the PhotoCapture Tutorial component.
+ */
+export enum TutorialSteps {
+  /**
+   * Welcome step.
+   */
+  WELCOME = 'welcome',
+  /**
+   * Guideline step.
+   */
+  GUIDELINE = 'guideline',
+  /**
+   * Sight tutorial step.
+   */
+  SIGHT_TUTORIAL = 'sightTutorial',
+  /**
+   * Sight step.
+   */
+  SIGHT = 'sight',
+}
+
+function getTutorialState(
+  enableTutorial: PhotoCaptureTutorialOption | undefined,
+): TutorialSteps | null {
+  const isFirstTime = !!localStorage.getItem(STORAGE_KEY_PHOTO_CAPTURE_TUTORIAL);
+
+  return enableTutorial === PhotoCaptureTutorialOption.DISABLED ||
+    (enableTutorial === PhotoCaptureTutorialOption.FIRST_TIME_ONLY && isFirstTime)
+    ? null
+    : TutorialSteps.WELCOME;
+}
+
+/**
+ * Parameters of the usePhotoCaptureTutorial hook.
+ */
+export interface PhotoCaptureTutorial
+  extends Pick<
+    CaptureAppConfig,
+    'enableTutorial' | 'enableSightTutorial' | 'enableSightGuidelines'
+  > {}
+
+/**
+ * Custom hook used to manage the state of photo capture tutorial.
+ */
+export function usePhotoCaptureTutorial({
+  enableTutorial,
+  enableSightTutorial,
+  enableSightGuidelines,
+}: PhotoCaptureTutorial) {
+  const [currentTutorialStep, setCurrentTutorialStep] = useState<TutorialSteps | null>(
+    getTutorialState(enableTutorial),
+  );
+
+  const goToNextTutorialStep = () => {
+    let steps = Object.values(TutorialSteps);
+    if (!enableSightGuidelines) {
+      steps = steps.filter((v) => v !== TutorialSteps.GUIDELINE);
+    }
+    if (!enableSightTutorial) {
+      steps = steps.filter((v) => v !== TutorialSteps.SIGHT_TUTORIAL);
+    }
+    if (currentTutorialStep === steps.at(-1)) {
+      setCurrentTutorialStep(null);
+      return;
+    }
+    const currentIndex = steps.findIndex((v) => currentTutorialStep === v);
+    setCurrentTutorialStep(steps[currentIndex + 1]);
+  };
+
+  const closeTutorial = () => {
+    setCurrentTutorialStep(null);
+  };
```

Comment:
Wrap both of these callbacks in a `useCallback` hook. We always wrap every function returned by a hook that will or can be used by a client with useCallback. For example in every hook of the common package etc. Usually, you're right, for hooks that will only be used by our own components and that are not exported in the package we don't have to unless we actually need it for the component. However the PhotoCapture component is a bit special because some clients might want to implement it themselves and use our logic to build their own PhotoCapture component. That's why every function returned by hooks like these should be wrapped in useCallback